sdk/clientcore/core/src/test/java/io/clientcore/core/util/binarydata/BinaryDataTest.java

```diff
@@ -74,6 +77,44 @@ public void fromCustomObject() throws IOException {
         assertEquals(expectedValue, data.toObject(expectedValue.getClass(), SERIALIZER));
     }
 
+    @Test
+    public void testToObjectFromType() throws IOException {
+        // Create a list of Person objects
+        final Person person1 = new Person().setName("John");
+        final Person person2 = new Person().setName("Jack");
+
+        List<Person> personList = new ArrayList<>();
+        personList.add(person1);
+        personList.add(person2);
+
+        // Convert the list to BinaryData
+        BinaryData binaryData = BinaryData.fromObject(personList);
+
+        // Use toObjectFromType to convert back to a list of Person objects
+        List<Person> persons = binaryData.toObjectFromType(new ParameterizedType() {
+            @Override
+            public Type[] getActualTypeArguments() {
+                return new Type[] { Person.class };
+            }
+
+            @Override
+            public Type getRawType() {
+                return List.class;
+            }
+
+            @Override
+            public Type getOwnerType() {
+                return null;
+            }
+        });
+
+        // Assert that the original and deserialized lists are equal
+        assertEquals(2, persons.size());
+        // Confirm java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class io.clientcore.core.models.Person
```

Comment:
@alzimmermsft is it expected to see this error?

Reply:
Shouldn't happen, looks to be a gap in handling. Without looking into it deeply, my guess is there is a gap with container types (List, Map, etc).